pkg/credentialprovider/config.go

```diff
@@ -122,14 +123,27 @@ func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error
 	if len(searchPaths) == 0 {
 		searchPaths = DefaultDockerConfigJSONPaths()
 	}
```

Comment:
if you want to extract the ability to load a single ConfigJSON file, then having `ReadDockerConfigJSONFile` continue looping over the search paths and calling `ReadSpecificDockerConfigJSONFile(filepath)` seems ok. That should be the only change needed to allow downstream to use the ReadSpecificDockerConfigJSONFile function to read a particular file from a mounted secret location.